src/smtencoding/FStar.SMTEncoding.Solver.fst

```diff
@@ -216,7 +217,9 @@ type query_settings = {
     query_errors:list errors;
     query_all_labels:error_labels;
     query_suffix:list decl;
-    query_hash:option string
+    query_hash:option string;
+    query_can_be_split_and_retried:bool;
```

Comment:
A query can't be retried if we're already in the retry loop or if split_queries is on